math/src/stableswap/math.rs

```diff
@@ -28,21 +29,23 @@ pub fn calculate_out_given_in<const D: u8, const Y: u8>(
 	idx_out: usize,
 	amount_in: Balance,
 	amplification: Balance,
+	pegs: &[(Balance, Balance)],
 ) -> Option<Balance> {
-	if idx_in == idx_out {
-		return None;
-	}
-	if idx_in >= initial_reserves.len() || idx_out >= initial_reserves.len() {
+	// Safety guards
+	let assets_ct = initial_reserves.len();
+	if assets_ct <= 1 || idx_in == idx_out || idx_in >= assets_ct || idx_out >= assets_ct || pegs.len() != assets_ct {
```

Comment:
ideally i would skip all peg related logic if there are no pegs

Reply:
there is always peg, if pool does not have pegs, it is all set to 1. 

initially, i had Option<Pegs> everywhere, but it looked very confusing, and introduces many IF THEN paths. It is not only list of pegs, but eg for Y calculation you nee to select a peg value that it is ommitted, that would have to an option too etc. 

with this approach there is no need to distinguish between pegs and no pegs.  And it does not introduce big overhead.